app/models/user.rb

```diff
@@ -1,6 +1,9 @@
 class User < ApplicationRecord
+  has_many :authored_tests, class_name: 'Test', dependent: :nullify
```

Comment:
такая запись будет искать в таблице `tests` внешний ключ `user_id` (модель же `User`)
у тебя же внешний ключ ссылающийся на автора - это `author_id`
поэтому нужно здесь явно указать по какому внешнему ключу ассоциация должна искать записи
```suggestion
  has_many :authored_tests, class_name: 'Test', foreign_key: :author_id, dependent: :nullify
```